devito/types/dimension.py

```diff
@@ -1187,6 +1255,38 @@ def _symbolic_thickness(cls, name):
                 Scalar(name="%s_rtkn" % name, dtype=np.int32, nonnegative=True))
 
 
+class StencilDimension(BasicDimension):
+
+    def __init_finalize__(self, name, size):
+        self._spacing = sympy.S.One
```

Comment:
may need start/end in case non-symmetric

Reply:
yepper, I have local edits that I haven't pushed yet which do exactly that